eksrollup/cli.py

```diff
@@ -229,11 +238,22 @@ def update_asgs(asgs, cluster_name):
                     terminate_instance_in_asg(outdated['InstanceId'])
                     if not instance_terminated(outdated['InstanceId']):
                         raise Exception('Instance is failing to terminate. Cancelling out.')
-
                     between_nodes_wait = app_config['BETWEEN_NODES_WAIT']
-                    if between_nodes_wait != 0:
-                        logger.info(f'Waiting for {between_nodes_wait} seconds before continuing...')
-                        time.sleep(between_nodes_wait)
+                    between_nodes_wait_pod_regex = app_config['BETWEEN_NODES_WAIT_POD_REGEX']
+                    if between_nodes_wait_pod_regex:
+                        try:
+                            between_nodes_wait_pod_regex_compiled = re.compile(rf'{between_nodes_wait_pod_regex}')
+                            logger.info(f'Regex sucessfully compiled: {between_nodes_wait_pod_regex}')
+                        except re.error:
+                            logger.error(f'{between_nodes_wait_pod_regex} is not a valid regex pattern!')
+                            exit(1)
+                        k8s_matched_pods = match_k8s_pods(k8s_node_pods, k8s_node_pods_list, between_nodes_wait_pod_regex_compiled)
+                        if len(k8s_matched_pods) > 0:
+                            logger.info(f'Waiting for k8s pods: {k8s_matched_pods}')
+                            while not pods_in_ready_state(k8s_matched_pods):
+                                wait_for_node(30)
+                    if len(k8s_matched_pods) == 0 and between_nodes_wait != 0:
+                        wait_for_node(between_nodes_wait)
```

Comment:
This code **is not executed** if ASG_USE_TERMINATION_POLICY is used...and guess what, I was using it for each run to avoid long waiting times until all instances are destroyed one by one...